server/web_ui/admin/src/components/admin_objectgraph.rs

```diff
@@ -0,0 +1,339 @@
+    }
+}
+
+impl TryFrom<String> for ObjectType {
+    type Error = ();
+
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        all::<ObjectType>()
+            .find(|x| format!("{x}") == value)
+            .ok_or(())
+    }
+}
+
+pub enum State {
+    Waiting,
+    Ready { entries: Vec<Entry> },
+    Error { emsg: String, kopid: Option<String> },
+}
+
+pub struct AdminObjectGraph {
+    state: State,
+    filters: Vec<ObjectType>,
+}
+
+impl Component for AdminObjectGraph {
+    type Message = Msg;
+    type Properties = ();
+
+    fn create(ctx: &Context<Self>) -> Self {
+        #[cfg(debug_assertions)]
+        console::debug!("views::objectgraph::create");
+
+        ctx.link()
+            .send_future(async { Self::fetch_objects().await.unwrap_or_else(|v| v.into()) });
+
+        let state = State::Waiting;
+
+        AdminObjectGraph {
+            state,
+            filters: vec![],
+        }
+    }
+
+    fn update(&mut self, _ctx: &Context<Self>, msg: Self::Message) -> bool {
+        #[cfg(debug_assertions)]
+        console::debug!("views::objectgraph::update");
+        match msg {
+            Msg::NewObjects { entries } => {
+                #[cfg(debug_assertions)]
+                console::debug!("Received new objects");
+                self.state = State::Ready { entries }
+            }
+            Msg::NewFilters { filters } => {
+                #[cfg(debug_assertions)]
+                console::debug!("Received new filters");
+                self.filters = filters;
+            }
+            Msg::Error { emsg, kopid } => self.state = State::Error { emsg, kopid },
+        }
+
+        true
+    }
+
+    fn changed(&mut self, _ctx: &Context<Self>, _props: &Self::Properties) -> bool {
+        #[cfg(debug_assertions)]
+        console::debug!("views::objectgraph::changed");
+        false
+    }
+
+    fn view(&self, ctx: &Context<Self>) -> Html {
+        match &self.state {
+            State::Waiting => self.view_waiting(),
+            State::Ready { entries } => self.view_ready(ctx, &self.filters, entries),
+            State::Error { emsg, kopid } => self.view_error(ctx, emsg, kopid.as_deref()),
+        }
+    }
+
+    fn rendered(&mut self, _ctx: &Context<Self>, _first_render: bool) {
+        #[cfg(debug_assertions)]
+        console::debug!("views::objectgraph::rendered");
+    }
+}
+
+impl AdminObjectGraph {
+    fn view_waiting(&self) -> Html {
+        loading_spinner()
+    }
+
+    fn view_ready(
+        &self,
+        ctx: &Context<Self>,
+        filters: &[ObjectType],
+        entries: &[Entry],
+    ) -> Html {
+        let typed_entries = entries
+            .iter()
+            .filter_map(|entry| {
+                let classes = entry.attrs.get("class")?;
+                let uuid = entry.attrs.get("uuid")?.first()?;
+
+                // Logic to decide the type of each entry
+                let obj_type = if classes.contains(&"group".to_string()) {
```

Comment:
@yaleman I guess we could also just add `class: Builtin` automatically to anything that is builtin? Makes it a bit easier to distinguish.

Reply:
Yeah, that would make sense

Reply:
@ToxicMushroom Can you open an issue that we need to add class: Builtin as a marker?